bar-assistant proxmox community script failing on version incompatibility

vue-salt-rim pins typescript@6.0.3 as a dev dependency, but one of its other dependencies, openapi-typescript@7.13.0, declares a peer dependency requiring typescript@^5.x. npm's default ERESOLVE behavior (npm 7+) refuses to install when it can't satisfy that peer constraint, so npm install fails with exit code 1 partway through the container setup script.

```
  ✔️  Installed Bar Assistant
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: vue-salt-rim@undefined
npm error Found: typescript@6.0.3
npm error node_modules/typescript
npm error   dev typescript@"6.0.3" from the root project
npm error
npm error Could not resolve dependency:
npm error peer typescript@"^5.x" from openapi-typescript@7.13.0
npm error node_modules/openapi-typescript
npm error   dev openapi-typescript@"7.13.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /root/.npm/_logs/2026-08-19T01_23_13_211Z-eresolve-report.txt
npm notice
npm notice New major version of npm available! 10.9.8 -> 12.0.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v12.0.2
npm notice To update run: npm install -g npm@12.0.2
npm notice
npm error A complete log of this run can be found in: /root/.npm/_logs/2026-08-19T01_23_13_211Z-debug-0.log
  ✖️  in line 71: exit code 1 (General error / Operation not permitted): while executing command  npm install

  --- Last 20 lines of log ---
[2026-08-18 20:22:10] [INFO] Installing PHP 8.4 packages
[2026-08-18 20:22:32] [OK] Setup PHP 8.4
[2026-08-18 20:22:32] [INFO] Setup Composer
[2026-08-18 20:22:34] [OK] Setup Composer
[2026-08-18 20:22:34] [INFO] Setup Node.js 22
[2026-08-18 20:22:41] [OK] Setup Node.js 22
[2026-08-18 20:22:42] [INFO] Setup MeiliSearch
[2026-08-18 20:22:42] [INFO] Fetching GitHub release: meilisearch (1.53.1)
[2026-08-18 20:22:47] [OK] Deployed: meilisearch (1.53.1)
[2026-08-18 20:22:49] [OK] Setup MeiliSearch 1.53.1
[2026-08-18 20:22:49] [INFO] Fetching GitHub release: bar-assistant (6.1.0)
[2026-08-18 20:22:50] [OK] Deployed: bar-assistant (6.1.0)
[2026-08-18 20:22:51] [INFO] Fetching GitHub release: vue-salt-rim (5.1.0)
[2026-08-18 20:22:52] [OK] Deployed: vue-salt-rim (5.1.0)
[2026-08-18 20:22:52] [INFO] Configuring PHP
[2026-08-18 20:22:52] [INFO] configured PHP
[2026-08-18 20:22:52] [INFO] Installing Bar Assistant
[2026-08-18 20:23:13] [OK] Installed Bar Assistant
[2026-08-18 20:23:13] [INFO] Installing Salt Rim
[2026-08-18 20:23:26] [ERROR] in line 71: exit code 1 (General error / Operation not permitted): while executing command  npm install
  -----------------------------------
```

--- a/ct/bar-assistant.sh
+++ b/ct/bar-assistant.sh
@@ -71,13 +71,15 @@ function update_script() {
     systemctl stop nginx
     msg_ok "Stopped nginx"
 
-    fetch_and_deploy_gh_release "vue-salt-rim" "karlomikus/vue-salt-rim" "tarball" "latest" "/opt/vue-salt-rim"
+    NODE_VERSION="22" NODE_MODULE="bun" setup_nodejs
+
+    CLEAN_INSTALL=1 fetch_and_deploy_gh_release "vue-salt-rim" "karlomikus/vue-salt-rim" "tarball" "latest" "/opt/vue-salt-rim"
     restore_backup
 
     msg_info "Configuring Vue Salt Rim"
     cd /opt/vue-salt-rim
-    $STD npm install
-    $STD npm run build
+    $STD bun install
+    $STD bun run build
     msg_ok "Configured Vue Salt Rim"
 
     msg_info "Starting nginx"

--- a/install/bar-assistant-install.sh
+++ b/install/bar-assistant-install.sh
@@ -25,7 +25,7 @@ msg_ok "Installed Dependencies"
 
 PHP_VERSION="8.4" PHP_FPM="YES" PHP_MODULE="pdo-sqlite" setup_php
 setup_composer
-NODE_VERSION="22" setup_nodejs
+NODE_VERSION="22" NODE_MODULE="bun" setup_nodejs
 setup_meilisearch
 fetch_and_deploy_gh_release "bar-assistant" "karlomikus/bar-assistant" "tarball" "latest" "/opt/bar-assistant"
 fetch_and_deploy_gh_release "vue-salt-rim" "karlomikus/vue-salt-rim" "tarball" "latest" "/opt/vue-salt-rim"
@@ -68,8 +68,8 @@ window.srConfig = {}
 window.srConfig.API_URL = "http://${LOCAL_IP}/bar"
 window.srConfig.MEILISEARCH_URL = "http://${LOCAL_IP}/search"
 EOF
-$STD npm install
-$STD npm run build
+$STD bun install
+$STD bun run build
 msg_ok "Installed Salt Rim"
 
 msg_info "Creating Service"

